### PR TITLE
Spring boot 01

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

File: src/main/resources/templates/index.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Index</title>
+</head>
+<body>
+<h1>Hello mates</h1>
+</body>
+</html>
```
Comment (yurwing) [inline src/main/resources/templates/index.html]: empty line

**Review by yurwing — APPROVED** [review]

File: src/main/java/com/example/boot/repository/RoleRepository.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.boot.repository;
+
+import com.example.boot.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = ?1")
+    Optional<Role> getByName(String name);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/boot/repository/RoleRepository.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Index</title>
```
Comment (KseniiaMakarova) [inline src/main/resources/templates/index.html]: do we need to push this file here?

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1 +1 @@
-
+spring.datasource.url=jdbc:h2:mem:test
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties]: let's enable access through console as well

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/boot/service/RoleService.java [hunk 5]
```diff
@@ -5,5 +5,5 @@
 public interface RoleService {
     Role save(Role role);
 
-    Role getByName(String name);
+    Role getByRoleName(Role.RoleName roleName);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/boot/service/RoleService.java]: you shouldn't change the method in service - we should be able to pass String on the service level - and here it should be transformed into RoleName type

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1 +1 @@
-spring.datasource.url=jdbc:h2:mem:test
+
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties]: let’s configure access to h2 database through the console in `application.properties`
you should be able to view your h2 DB from browser

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**